mindsdb/interfaces/chatbot/memory.py

```diff
@@ -132,15 +132,32 @@ def _get_chat_history(self, chat_id, table_name):
             )
         )
 
+        # Convert the WHERE conditions to a BinaryOperation objects.
+        # If there are more than two conditions, nest them in a BinaryOperation objects.
+        if len(where_conditions) > 2:
+            where_conditions_binary_operation = BinaryOperation(
+                op='and',
+                args=where_conditions[:2]
+            )
+
+            for condition in where_conditions[2:]:
+                where_conditions_binary_operation = BinaryOperation(
+                    op='and',
+                    args=[where_conditions_binary_operation, condition]
+                )
+        # If there are only two conditions or less, use them as is.
```

Comment:
it has to be exactly two conditions (not less, not more) to use in BinaryOperation

Simply all it could be done by:
```python
where_conditions_binary_operation = None
for condition in where_conditions:
        if where_conditions_binary_operation is None:
              where_conditions_binary_operation = condition
        else:
              where_conditions_binary_operation = BinaryOperation('and', args=[where_conditions_binary_operation, condition])
```

Reply:
@ea-rus Sorry for the confusion. Please check now.